lib/osutil/atomic.go

```diff
@@ -92,7 +93,15 @@ func (w *AtomicWriter) Close() error {
 		return err
 	}
 
-	if err := w.fs.Rename(w.next.Name(), w.path); err != nil {
+	err := w.fs.Rename(w.next.Name(), w.path)
+	if runtime.GOOS == "windows" && fs.IsPermission(err) {
+		// On Windows, we might not be allowed to rename over the file
+		// because it's read-only. Get us some write permissions and try
+		// again.
+		_ = w.fs.Chmod(w.path, 0644)
```

Comment:
I'm not gonna try to do a stat and defer an undo of this. We're supposed to overwrite and destroy this file, with a file that we *have* write permissions on, and if it didn't help to set the old file to writeable then who knows what's even happening any more